src/python/pants/goal/run_tracker.py

```diff
@@ -447,24 +447,22 @@ def run_information(self):
     return run_information
 
   def _stats(self):
+    stats = {
+      'run_info': self.run_information(),
+      'artifact_cache_stats': self.artifact_cache_stats.get_all(),
+      'pantsd_stats': self.pantsd_stats.get_all(),
+      'cumulative_timings': self.cumulative_timings.get_all(),
+      'recorded_options': self._get_options_to_record(),
+    }
     if self._stats_version == 2:
-      return {
-        'run_info': self.run_information(),
-        'artifact_cache_stats': self.artifact_cache_stats.get_all(),
-        'pantsd_stats': self.pantsd_stats.get_all(),
-        'workunits': self.json_reporter.results,
-      }
+      stats['workunits'] = self.json_reporter.results
     else:
-      return {
-        'run_info': self.run_information(),
-        'cumulative_timings': self.cumulative_timings.get_all(),
+      stats.update({
         'self_timings': self.self_timings.get_all(),
-        'critical_path_timings': self.get_critical_path_timings().get_all(),
-        'artifact_cache_stats': self.artifact_cache_stats.get_all(),
-        'pantsd_stats': self.pantsd_stats.get_all(),
+        'critical_path_timings': self.get_critical_path_timings().get_all(),        
         'outcomes': self.outcomes,
-        'recorded_options': self._get_options_to_record(),
-      }
+      })
+      return stats
```

Comment:
I believe this return should be outside the conditional.

Reply:
Good catch!

Reply:
🤦‍♂